[#63744] Make Project Settings form subheadings go full width

myabc opened the pull request:
# Ticket

https://community.openproject.org/wp/63744

# What are you trying to accomplish?

Ensure subheadings take up full page width.

## Screenshots

| Before | After |
|--------|--------|
| <img width="1502" height="778" alt="Screenshot 2025-12-26 at 14 41 10" src="https://github.com/user-attachments/assets/2b685e72-23a5-40fc-949c-d3ba36d6d42d" /> | <img width="1508" height="785" alt="Screenshot 2025-12-26 at 14 40 45" src="https://github.com/user-attachments/assets/5510dd8f-0251-4b61-a097-80db378b59da" /> | 

# What approach did you choose and why?

This PR moves the subheadings (`Primer::Beta::Subhead`) outside of each of the individual forms, to ensure they take up the full width of the page.

I also removed the vertical flex layout, instead dividing the page using `section` tags with corresponding margins. This is both for good semantics and to ensure the feature specs (which rely on `within_section` from Capybara accessible selectors) still pass.

# Merge checklist

- [ ] Added/updated tests
- [ ] Added/updated documentation in Lookbook (patterns, previews, etc)
- [x] Tested major browsers (Chrome, Firefox, Edge, ...)

myabc committed:
[#63744] Make Settings form subheadings full width

https://community.openproject.org/work_packages/63744

cbliard reviewed (approved):
This works 🥂

While you're at it, could you also make the space between section be 24px like on the figma?

Currently it's 16px:

<img width="336" height="147" alt="image" src="https://github.com/user-attachments/assets/9b9a60ae-b943-43b5-9e72-0b40a0f5dc27" />

On [Figma designs](https://www.figma.com/design/k8FvtPsyKrbqII2tHg0M0q/Project-Information?node-id=0-1&p=f&t=3ZMJ2iB1ITiUozRe-0) 24px is specified:

<img width="876" height="225" alt="image" src="https://github.com/user-attachments/assets/d7efd50f-5575-4bea-ae56-4f3bc73872bd" />

I always found the sections to be too close from each other. I'm glad that the specification actually gives them more space.

myabc committed:
Increase vertical spacing to 24px (per Figma)

Increases bottom margin on `section`s to conform to original Figma
designs. Also increases banner bottom margin to 16px.

HT / @cbliard
